### ISSUE TITLE
Wont compile for ODIN using OdinWifiInterface

### ISSUE THREAD

**Issue by andcor02** [ARMmbed/oob-mbed-os-example-client#13]
Trying to compile for ODIN platform and use OdinWifiInterface.

Compiling for GCC_ARM.

Receive the error:

[ERROR] ./mbed-os/targets/TARGET_STM/TARGET_STM32F4/TARGET_UBLOX_EVK_ODIN_W2/sdk/TOOLCHAIN_GCC_ARM/libublox-odin-w2-driver.a(cb_md5.o): In function `cbMD5_md5Vector':
cb_md5.c:(.text.cbMD5_md5Vector+0x10): undefined reference to `mbedtls_md5_starts'
cb_md5.c:(.text.cbMD5_md5Vector+0x24): undefined reference to `mbedtls_md5_update'
cb_md5.c:(.text.cbMD5_md5Vector+0x2e): undefined reference to `mbedtls_md5_finish'
collect2: error: ld returned 1 exit status

**Comment by MarceloSalazar** [comment]
Please confirm version of gcc

**Comment by bulislaw** [comment]
@andreaslarssonublox that's new have you seen it before?

**Comment by andcor02** [comment]
If anyone can check this actually works with ODIN that'd be great. Regardless of wifi working at this point, just want to make sure this can compile at the least

**Comment by bulislaw** [comment]
Will do.

**Comment by andreaslarssonublox** [comment]
I think the problem is that the MBEDTLS_USER_CONFIG_FILE is used both in the ODIN driver target definition and the example. I think we can solve it in two ways(at least):
1. Remove the mbedtls file for the ODIN target and put it in as macros instead
2. Put the defines in the medtls config of the example

What do you think @bulislaw @MarceloSalazar ?

**Comment by bulislaw** [comment]
Yeah just got to the same conclusion: https://github.com/ARMmbed/oob-mbed-os-example-client/pull/4

**Comment by bulislaw** [comment]
@0xc0170 I've tried to overwrite it yet once again in the mbed_app.json, but it still doesn't work it seems we can only have it defined once. Any ideas? Maybe adding it to README would be enough?

**Comment by andreaslarssonublox** [comment]
@andcor02 @MarceloSalazar 
Can you see if the following PRs fixes your problem(both should be used)?
https://github.com/ARMmbed/mbed-os/pull/3054
https://github.com/ARMmbed/mbed-os/pull/3053

**Comment by andcor02** [comment]
Now complies correctly with GCC_ARM.

However, connect to wireless network. Unable to connect to mbed device connector.

**Comment by 0xc0170** [comment]
cc @sbutcher-arm @yanesca Please have a look

**Comment by andreaslarssonublox** [comment]
The compiler error is there because the ODIN driver uses those deprecated functions. We can in the next release adapt to not using the deprecated functions but there is not time to do it now I guess. 
What do you think @sbutcher-arm?
Should we instead put the defines in the mbedtls_mbed_client_config.h as mentioned as nr 2 above?

**Comment by ciarmcom** [comment]
ARM Internal Ref: IOTCLT-1119

**Comment by MarceloSalazar** [comment]
Should be fixed in RC4

**Comment by sbutcher-arm** [comment]
@andreaslarssonublox - The ODIN specific configurations don't belong in `mbedtls_mbed_client_config.h` as they will apply to *all* targets, not just ODIN. Best keep them where they are.
